homeassistant/components/bmw_connected_drive/coordinator.py

```diff
@@ -56,20 +56,19 @@ async def _async_update_data(self) -> None:
 
         try:
             await self.account.get_vehicles()
-        except (HTTPError, HTTPStatusError, TimeoutException) as err:
-            if isinstance(err, HTTPStatusError) and err.response.status_code == 429:
-                # Increase scan interval to not jump to not bring up the issue next time
-                self.update_interval = timedelta(
-                    seconds=DEFAULT_SCAN_INTERVAL_SECONDS * 3
+        except MyBMWAuthError as err:
+            # Clear refresh token and trigger reauth
+            self._update_config_entry_refresh_token(None)
+            raise ConfigEntryAuthFailed(str(err)) from err
+
+        except (MyBMWAPIError, RequestError) as err:
+            if self.last_update_success is True:
+                _LOGGER.warning(
+                    "Error communicating with BMW API (%s): %s",
+                    type(err).__name__,
+                    err,
                 )
-            if isinstance(err, HTTPStatusError) and err.response.status_code in (
-                401,
-                403,
-            ):
-                # Clear refresh token only and trigger reauth
-                self._update_config_entry_refresh_token(None)
-                raise ConfigEntryAuthFailed(str(err)) from err
-            raise UpdateFailed(f"Error communicating with BMW API: {err}") from err
+                self.last_update_success = False
```

Comment:
Why are we not raising `UpdateFailed` which will also do the logging?

Reply:
Because I wasn't aware (didn't check the `UpdateFailed` logic) and I saw it implemented in another component. Updated it.